src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -2391,6 +2394,30 @@ GUI_TEST_CLASS_DEFINITION(test_7548) {
     }
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7550) {
+    auto clickFindProducts = [&os] {
+        if (GTUtilsMdi::activeWindow(os)->objectName() != "123 [cant_translate.fa]") {
+            GTUtilsMdi::closeActiveWindow(os);
+        }
+        GTUtilsOptionPanelSequenceView::pressFindProducts(os);
+    };
+    // Open the _common_data/fasta/cant_translate.fa.
+    // Open the "In Silico PCR" tab.
+    // Set "AAAAAAAAAAAAAAA" as forward and reverse primers.
+    // Click "Find product(s) anyway" 100 times.
+    // Click "Find product(s) anyway" again.
+    //     Expected state: the number of notifications is "99+", there is no crash.
+    GTUtilsMdi::closeActiveWindow(os);
+    GTFileDialog::openFile(os, testDir + "_common_data/fasta/cant_translate.fa");
+    GTUtilsSequenceView::checkSequenceViewWindowIsActive(os);
+    GTUtilsOptionPanelSequenceView::setForwardPrimer(os, "AAAAAAAAAAAAAAA");
+    GTUtilsOptionPanelSequenceView::setReversePrimer(os, "AAAAAAAAAAAAAAA");
+    for (auto notificationStack = AppContext::getMainWindow()->getNotificationStack(); notificationStack->count() < 100;
```

Comment:
Do not access UI state from the test thread: this is unsafe because UI model changes concurrently from another (UI) thread. This is one of the main reason why our tests are flaky.

You need to add helper methods need with 'runInMainThread()' around the UI accessors call them from the test method.

Reply:
Main thread scenario added, please check the correctness.